### PR TITLE
[EVM] add `wasmd` precompile

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Add a `wasmd` precompile that allows EVM contracts to access the following CosmWasm functions for any CW contract:
- `instantiate`
- `execute`
- `query`

## Testing performed to validate your change
unit test

### CODE REVIEW

File: evmrpc/state_test.go [hunk 1]
```diff
@@ -193,46 +191,3 @@ func TestGetStorageAt(t *testing.T) {
 		})
 	}
 }
-
-func TestGetProof(t *testing.T) {
```
Comment (jewei1997) [inline evmrpc/state_test.go:197]: why was this deleted?

File: example/cosmwasm/echo/src/contract.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#[cfg(not(feature = "library"))]
```
Comment (jewei1997) [inline example/cosmwasm/echo/src/contract.rs:1]: what are these cosmwasm contracts for?

File: testutil/keeper/evm.go [hunk 3]
```diff
@@ -3,82 +3,37 @@ package keeper
...
-	"github.com/sei-protocol/sei-chain/x/evm/types"
 )
 
-func MockEVMKeeper() (*evmkeeper.Keeper, *paramskeeper.Keeper, sdk.Context) {
-	evmStoreKey := sdk.NewKVStoreKey(types.StoreKey)
-	authStoreKey := sdk.NewKVStoreKey(authtypes.StoreKey)
-	bankStoreKey := sdk.NewKVStoreKey(banktypes.StoreKey)
-	stakingStoreKey := sdk.NewKVStoreKey(stakingtypes.StoreKey)
-	keyParams := sdk.NewKVStoreKey(typesparams.StoreKey)
-	tKeyParams := sdk.NewTransientStoreKey(typesparams.TStoreKey)
```
Comment (jewei1997) [inline testutil/keeper/evm.go:37]: why are these no longer needed?

**Review by jewei1997 — APPROVED** [review]
very cool, LGTM!

File: precompiles/setup.go [hunk 4]
```diff
@@ -1,29 +1,50 @@
...
+	addPrecompileToVM(bankp, bankp.Address())
+	wasmdp, err := wasmd.NewPrecompile(evmKeeper, wasmdKeeper, wasmdViewKeeper)
+	if err != nil {
+		return err
+	}
+	addPrecompileToVM(wasmdp, wasmdp.Address())
 	return nil
 }
+
+// This function modifies global variable in `vm` module. It should only be called once
```
Comment (philipsu522) [inline precompiles/setup.go:41]: should we follow a singleton pattern here to ensure it's only initialized once (or something like baseapp.sealed where it panics if called multiple times)?

File: precompiles/wasmd/wasmd_test.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	evm := vm.EVM{
+		StateDB: statedb,
+	}
+	res, err := p.Run(&evm, append(p.QueryID, args...))
+	require.Nil(t, err)
+	outputs, err := queryMethod.Outputs.Unpack(res)
+	require.Nil(t, err)
+	require.Equal(t, 1, len(outputs))
+	require.Equal(t, "{\"message\":\"query test\"}", string(outputs[0].([]byte)))
+}
```
Comment (philipsu522) [inline precompiles/wasmd/wasmd_test.go:187]: thoughts on adding more test cases around failure/edge cases? Most of these look happy path, and the added tests would help ensure wasmd.go doesn't have different behavior when we modify

File: evmrpc/state_test.go (hunk 1 above)
Comment (codchen) [reply evmrpc/state_test.go:197]: there was a bug in how stores in tests were initialized that were fixed in this PR, and apparently GetProof would only work with the presence of that bug, so GetProof is in reality broken. I've added a comment in GetProof to fix it in a later PR given this one is already too large.

File: testutil/keeper/evm.go (hunk 3 above)
Comment (codchen) [reply testutil/keeper/evm.go:37]: they are initialized more properly via `testApp := Setup(false)` above

File: precompiles/setup.go (hunk 4 above)
Comment (codchen) [reply precompiles/setup.go:41]: made it panic if already initialized

File: precompiles/wasmd/wasmd_test.go (hunk 5 above)
Comment (codchen) [reply precompiles/wasmd/wasmd_test.go:187]: yeah we should

Comment (codchen) [reply precompiles/wasmd/wasmd_test.go:187]: added
